src/profile-logic/profile-data.js

```diff
@@ -158,34 +158,19 @@ export function getFuncIndex(path: CallNodePath): IndexIntoFuncTable {
   return path[path.length - 1];
 }
 
-/**
- * This function Returns the JS implementation information for a specific stack.
- */
-export function getJsImplementationForStack(
-  stackIndex: IndexIntoStackTable,
-  { stackTable, frameTable, stringTable }: Thread
-): string {
-  const frameIndex = stackTable.frame[stackIndex];
-  const jsImplementationStrIndex = frameTable.implementation[frameIndex];
-  const jsImplementation =
-    jsImplementationStrIndex === null
-      ? 'interpreter'
-      : stringTable.getString(jsImplementationStrIndex);
-
-  return jsImplementation;
-}
-
-type Breakdown = { [key: string]: Milliseconds } | null;
+type JsImplementation = 'interpreter' | 'ion' | 'baseline' | 'unknown';
+type StackImplementation = 'native' | JsImplementation;
```

Comment:
Thanks for the added precision!

Reply:
Thanks for basically giving me the code ;)